friends/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import { Route, withRouter } from 'react-router-dom';
+import axios from 'axios';
+
+import FriendList from './components/FriendList';
+import NewFriend from './components/NewFriend';
+import EditFriend from './components/EditFriend';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+        friends: [],
+        currentFriend: []
```

Comment:
I might try and rework your `currentFriend` property, and use the raw properties of `name`, `age`, and `email` so you can avoid adding another level of nesting to your data.